test/evergreen/tcmalloc_install_or_build.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+#
+# Make a tcmalloc shared library avilable for preloading into the environment.
+
+set -u
+
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 build-variant"
+    exit 1
+fi
+
+build_variant=$1
+
+# This is tcmalloc upstream revision 093ba93 source patched by the SERVER team.
+# https://github.com/mongodb-forks/tcmalloc/releases/tag/mongo-SERVER-85737
+PATCHED_SRC=mongo-SERVER-85737
+PATCHED_TGZ="${PATCHED_SRC}.tar.gz"
+PATCHED_TGZ_URL="https://github.com/mongodb-forks/tcmalloc/archive/refs/tags/${PATCHED_TGZ}"
+PATCHED_SRC_DIR="tcmalloc-${PATCHED_SRC}"
+
+# Sensitive variables retrieved through Evergreen so they won't be exposed
+# in source control.
+#
+# Export the AWS_* variables so that they are visible for the 'aws' cli.
+S3_URL=${s3_bucket_tcmalloc}
+export AWS_ACCESS_KEY_ID=${s3_access_key}
+export AWS_SECRET_ACCESS_KEY=${s3_secret_key}
```

Comment:
Alternatively, we could use the `env` parameter in the Evergreen `shell.exec` command (to map the variable names) and save these 2 lines here. But I can see the benefit of explicitly setting the 2 x AWS_* variables here, closer to where the aws cli commands are used.

Reply:
I see the efficiency in what you are saying, but this is explicit and explained (comment), which I think is +ve in this circumstance.